internal/state/state.go

```diff
@@ -2,42 +2,65 @@ package state
 
 import (
 	"context"
-	"errors"
+	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
 	"path/filepath"
+	"time"
 
+	"github.com/hetznercloud/cli/internal/hcapi2"
+	"github.com/hetznercloud/cli/internal/version"
 	"github.com/hetznercloud/hcloud-go/v2/hcloud"
 )
 
-type State struct {
+type State interface {
+	TokenEnsurer
+	ActionWaiter
+	context.Context
+	hcapi2.Client
+
+	WriteConfig() error
+	Config() *Config
+}
```

Comment:
```suggestion
type State interface {
	context.Context

	TokenEnsurer
	ActionWaiter

	Client() hcapi2.Client
	Config() *Config
	// WriteConfig() error => move this to the Config struct e.g. Config().Write(args), not the state responsability
}
```

After looking at the changes, I think it is a good idea if we add a bit of name spacing here. The config and client APIs are quite verbose, while they can be confined and stand on their own.

Our state looks a lot like an extended context, so I moved the context to the top.

And I think it's alright if we leave the command helpers without namespace, I like the idea of injecting other interfaces in the state. We can implement functions that require e.g. the `TokenEnsurer` interface, and still be able to pass the entire State.